federation/pkg/federation-controller/replicaset/replicasetcontroller_test.go

```diff
@@ -180,6 +175,81 @@ func TestReplicaSetController(t *testing.T) {
 	assert.Equal(t, rs.Status.AvailableReplicas, rs1.Status.AvailableReplicas+rs2.Status.AvailableReplicas)
 }
 
+func TestReplicaSetControllerPuppets(t *testing.T) {
+
+	replicaSetReviewDelay = 10 * time.Millisecond
+	clusterAvailableDelay = 20 * time.Millisecond
+	clusterUnavailableDelay = 60 * time.Millisecond
+	allReplicaSetReviewDelay = 120 * time.Millisecond
+
+	fedclientset := fedclientfake.NewSimpleClientset()
+	fedrswatch := watch.NewFake()
```

Comment:
I'm for rewriting this test from scratch/significantly refactor. I would be easier if these two tests use the same names.